### PR TITLE
T30974

### PR DESCRIPTION
**Pull request by mschiu77**
Fix the compatibility problem caused by the f"" string usage and subprocess.run

### CODE REVIEW

**Comment by jprvita** [comment]
- Please split the mask / unmask changes into its own separate commit -- having each commit implementing only one thing (as much as possible) makes the review process a lot easier. And on those changes, let's not change the `metrics_unit` API -- let's keep the callers passing "start" or "stop", and encapsulate the extra necessary steps in `metrics_unit`'s implementation.
- The changes from f-strings to .format look good, except that they introduce some long lines that are likely to break the coding style. Please run each commit through [black](https://github.com/psf/black) and clean any warnings.
- The changes from `.run` to `.call` look good to me, but I haven't tested them yet.
- Finally, please make sure to always add a link to the phabricator task on each commit message, and on the PR description, so our phabricator integrations work correctly.

**Review by jprvita — CHANGES_REQUESTED** [review]
- Please split the mask / unmask changes into its own separate commit -- having each commit implementing only one thing (as much as possible) makes the review process a lot easier. And on those changes, let's not change the `metrics_unit` API -- let's keep the callers passing "start" or "stop", and encapsulate the extra necessary steps in `metrics_unit`'s implementation.
- The changes from f-strings to .format look good, except that they introduce some long lines that are likely to break the coding style. Please run each commit through [black](https://github.com/psf/black) and clean any warnings.
- The changes from `.run` to `.call` look good to me, but I haven't tested them yet.
- Finally, please make sure to always add a link to the phabricator task on each commit message, and on the PR description, so our phabricator integrations work correctly.

**Comment by mschiu77** [comment]
I do the mask/unmask thing in another T30975. I didn't notice that it's also in this this PR. Thanks for pointing that out.

File: tools/eos-metrics-collector.exe [hunk 1]
```diff
@@ -234,12 +235,12 @@ def collect(storage_root):
     data_size = collector.calculate_metrics_data_size()
     if free_space < data_size:
         msg = (
-            f"Insufficient free space ({free_space} bytes) for metrics "
-            f"data size ({data_size} bytes), aborting."
+            "Insufficient free space ({} bytes) for ".format(free_space) +
+            "metrics data size ({} bytes), aborting.".format(data_size)
```
Comment (jprvita) [inline tools/eos-metrics-collector.exe:239]:
```suggestion
            "Insufficient free space ({} bytes) for metrics data size "
            "({} bytes), aborting.".format(free_space, data_size)
```

**Review by jprvita — CHANGES_REQUESTED** [review]
I think you missed what black suggested -- when I run it on your latest branch it still changed those strings you are concatenating with `+`. But I suggest getting rid of the `+` altogether and only calling `.format()` once per string, as in the inline suggestions here.

But please apply them directly to the commit introducing the change (the f-string one) instead of accepting them in github, which I think will create a fixup commit on the tip of the branch instead. This keep the history a bit cleaner.

**Review by jprvita — APPROVED** [review]

**Pull request merged**